src/stackwalk.c

```diff
@@ -84,6 +84,84 @@ size_t rec_backtrace(uintptr_t *data, size_t maxsize)
     return rec_backtrace_ctx(data, maxsize, &context);
 }
 
+int jl_num_frames(uintptr_t ip, int skipC)
+{
+    jl_frame_t *frames;
+    int n = jl_getFunctionInfo(&frames, ip, skipC, 0);
+    int count = 0;
+    if (skipC) {
+        for (int i = 0; i < n; i++) {
+            jl_frame_t frame = frames[i];
+            if (!frame.fromC) count++;
+            free(frame.func_name);
+            free(frame.file_name);
+        }
+    }
+    else {
+        count = n;
+    }
+    free(frames);
+    return count;
+}
+
+int jl_frame_contains(uintptr_t ip, jl_sym_t *name, int skipC)
+{
+    if (!name)
+        return 0;
+
+    char *func_name = jl_symbol_name(name);
+    jl_frame_t *frames;
+    int n = jl_getFunctionInfo(&frames, ip, 0, 0);
+    int found = 0;
+    for (int i = 0; i < n && !found; i++) {
+        jl_frame_t frame = frames[i];
+        if (!skipC || (skipC && !frame.fromC)) {
+            if (frame.func_name && strcmp(frame.func_name, func_name) == 0)
+                found = 1;
+        }
+        free(frame.func_name);
+        free(frame.file_name);
+    }
+    free(frames);
+    return found;
+}
+
+JL_DLLEXPORT jl_value_t *jl_backtrace_caller(jl_sym_t *name, int num_callers)
+{
+    if (num_callers < 0)
+        return NULL;
+
+    jl_value_t *ip = (jl_value_t*)jl_svec2(jl_voidpointer_type, jl_box_long(1));
```

Comment:
Thanks, seems obvious now. I think I was worried that if I didn't allocate memory with julia that I could get left with a dangling pointer.